lib/parser.dart

```diff
@@ -8,8 +8,18 @@ class SvgParser {
   /// Parses SVG from a string to a [DrawableRoot].
   ///
   /// The [key] parameter is used for debugging purposes.
-  Future<DrawableRoot> parse(String str, {String? key}) async {
-    final SvgParserState state = SvgParserState(xml.parseEvents(str), key);
+  ///
+  /// The [dryRun] detects if an SVG contains contains unsupported features.
+  /// If true the function will throw with an error.
+  /// If false only warnings are logged to the console.
+  /// Defaults to false.
+  Future<DrawableRoot> parse(
+    String str, {
+    String? key,
+    bool dryRun = false,
```

Comment:
Here and elsewhere: I wonder if we can come up with a better name than `dryRun`. Seeing this pop up in my IDE as a suggestion doesn't immediately suggest anything to me.

Maybe something like `bool warningsAsErrors`, or `bool unhandledElementsAsErrors`?

Reply:
Makes sense to me:
I would go with `warningsAsErrors` as in the future the library might also warn about unsupported SVG attributes which is currently missing, but might be helpful.